mfkdf2/src/rng.rs

```diff
@@ -0,0 +1,59 @@
+#[cfg(feature = "differential-test")]
+pub mod det_rng {
+  use std::cell::RefCell;
+
+  use rand::{CryptoRng, RngCore, SeedableRng};
+  use rand_chacha::ChaCha20Rng;
+
+  const DEFAULT_SEED: [u8; 32] = [10u8; 32];
+  thread_local! {
+    static RNG: RefCell<ChaCha20Rng> = RefCell::new(ChaCha20Rng::from_seed(DEFAULT_SEED));
+  }
```

Comment:
this is cursed rust.

Can you use a `OnceCell` or whatever here to create a static? 

This clearly works but I've never seen it done this way.

Reply:
OnceCell can't work due to no mutable reference in static.
Other options would be a lazylock, but that would require a mutex. I used this to prevent any locks.

Reply:
https://docs.rs/rand/latest/src/rand/rngs/thread.rs.html#118-127

that's how rust rand do it as well.